activesupport/lib/active_support/cache/mem_cache_store.rb

```diff
@@ -154,11 +154,15 @@ def [](version)
 
           module Loader
             def load(payload)
-              if payload.is_a?(Entry)
-                payload
-              else
-                Cache::Coders::Loader.load(payload)
-              end
+              return payload if payload.is_a?(Entry)
+
+              # At this point, we expect payload to be a String in Marshal format.
+              # You should use the `raw:` option if you know this is not the case. But this may
+              # happen in cases you can't control, eg. if upgrading from an old version of
+              # the dalli_store. We return an Entry as a fallback.
+              return Entry.new(payload) unless Cache::Coders::Loader.valid_format?(payload)
```

Comment:
Please just check `is_a?(String)` here. 

In the future if someone were to upgrade incorrectly (e.g deploy new Rails version using the new format immediately), it's important that we fail rather than to return the serialized payload as a value.

Reply:
That won't work, this test will still fail if we just check for a string: https://github.com/rails/rails/pull/42566/files#diff-14cab604508ced2a334cad4623993cac14e1d4b55d6b4e1bc681b830924f492aR235-R240

Reply:
Your point is valid too 🤔

Reply:
I see. Then rather than raise an `ArgumentError`, we should return `nil`. It's actually better for people messing their upgrades, they'll just get a few misses. Maybe with a `Rails.logger.warn()`?

Reply:
Yeah, nils are better than cache reads crashing. I'll upgrade it to do that in all cases where it would currently raise.